akka-docs-dev/rst/scala/code/docs/http/scaladsl/server/FileUploadExamplesSpec.scala

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package docs.http.scaladsl.server
+
+import java.io.File
+
+import akka.http.scaladsl.model.Multipart.FormData.BodyPart
+import akka.stream.io.SynchronousFileSink
+import akka.stream.scaladsl._
+import akka.http.scaladsl.model.Multipart
+
+import scala.concurrent.Future
+
+class FileUploadExamplesSpec extends RoutingSpec {
+
+  //#simple-upload
+  val simpleRoute =
+    path("upload") {
+      entity(as[Multipart.FormData]) { formData =>
+        val completed: Source[File, _] = formData.parts.mapAsync(1) {
+          case BodyPart(name, entity, _, _) =>
+            val file = File.createTempFile("uploaded", "bin")
+            val sink = SynchronousFileSink(file)
+            entity.dataBytes.runWith(sink).map(_ => file)
+        }
```

Comment:
Yes and no :smile: 

Yes that's the "simple to file" case, it should be documented that's true.
I also think this is too much ceremony and we should provide a directive for it, for the simple case `path()(simpleFileUpload(to:File)(onUploaded:Try[File] => Response))`

The simple one is important, but we also should document the awesome part – while the data is incoming do `dataBytes.via(Framing.delimiter("\n").map(readCsv).map(convert).to(Sink.actorRef(something))`

so it's *really streaming* and actually doing stuff to the data while its streaming :)